Failed only in x86_64-darwin

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 368072`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hugo</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>hugo</li>
  </ul>
</details>

```
Running phase: checkPhase
@nix { "action": "setPhase", "phase": "checkPhase" }
go: downloading github.com/bep/helpers v0.5.0
--- FAIL: TestCommands (0.00s)
    --- FAIL: TestCommands/server__error_recovery_edit_config (6.81s)
        testscript.go:584: # Test the hugo server command when adding an error to a config file
            # and then fixing it. (6.796s)
            > hugo server &
            > waitServer
            > httpget ${HUGOTEST_BASEURL_0}p1/ 'Title: P1'
            > replace $WORK/hugo.toml 'title =' 'titlefoo'
            > httpget ${HUGOTEST_BASEURL_0}p1/ 'failed'
            FAIL: testscripts/commands/server__error_recovery_edit_config.txt:11: timeout waiting for "http://localhost:1313/p1/": response body "<script src=\"/livereload.js?mindelay=10&am
p;v=2&amp;port=1313&amp;path=livereload\" data-no-instant defer></script>Title: P1|BaseURL: http://localhost:1313/|\n" for URL "http://localhost:1313/p1/" does not match "failed"
            [background] hugo server: exit status 0
            [stdout]
            Watching for changes in $WORK/{content,layouts,static}
            Watching for config changes in $WORK/hugo.toml
            Start building sites …
            hugo v0.140.1+extended+withdeploy darwin/amd64 BuildDate=unknown VendorInfo=nixpkgs


                               | EN
            -------------------+-----
              Pages            |  4
              Paginator pages  |  0
              Non-page files   |  0
              Static files     |  1
              Processed images |  0
              Aliases          |  0
              Cleaned          |  0

            Built in 4 ms
            Environment: "development"
            Serving pages from disk
            Running in Fast Render Mode. For full rebuilds on change: hugo server --disableFastRender
            Web Server is available at http://localhost:1313/ (bind address 127.0.0.1)
            Press Ctrl+C to stop

FAIL
FAIL    github.com/gohugoio/hugo    17.939s
FAIL

```

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hugo</li>
  </ul>
</details>

--- a/pkgs/by-name/hu/hugo/package.nix
+++ b/pkgs/by-name/hu/hugo/package.nix
@@ -23,10 +23,18 @@ buildGoModule rec {
 
   vendorHash = "sha256-swcj1JxYoRqKscu/IC0uiAATp4AXN0aANWkSq/mJsyc=";
 
-  checkFlags = [
-    # Workaround for "failed to load modules"
-    "-skip=TestCommands/mod"
-  ];
+  checkFlags =
+    let
+      skippedTestPrefixes = [
+        # Workaround for "failed to load modules"
+        "TestCommands/mod"
+        # Server tests are flaky, at least in x86_64-darwin. See #368072
+        # We can try testing again after updating the `httpget` helper
+        # ref: https://github.com/gohugoio/hugo/blob/v0.140.1/main_test.go#L220-L233
+        "TestCommands/server"
+      ];
+    in
+    [ "-skip=^${builtins.concatStringsSep "|^" skippedTestPrefixes}" ];
 
   proxyVendor = true;
 

